### ISSUE TITLE
Filtering aggregations and adding nested aggregations

### ISSUE THREAD

**Issue by Locustv2** [yiisoft/yii2-elasticsearch#91]
When adding an aggregation to a query which has some conditions, the conditions are not applied to the aggregations.

For example:

`$results = Topics::find()->where(['authorId' => 123])->addAgg('categories', 'terms', ['field' => 'category'])->search();`

The `hits` return all records that matches the `where()` conditions but the aggregations does not consider the conditions. I think it would be better if the `filters` are also passed to the aggregations.

And also i could not find a direct way to do nested aggregations. I had to do it manually by modifying the `\yii\elasticsearch\ActiveQuery::$aggregations`

For example to achieve something like this:

```php
$this->aggregations = [
    'rating' => [
        'filter' => ['term' => ['authorId' => 123]],
        'aggregations' => [
            'categories' => [
                'terms' => [
                    'field' => 'category',
                    'size' => 0,
                ],
                'aggregations' => [
                    'avgRating' => [
                        'avg' => [
                            'field' => 'rating',
                        ]
                    ],
                ]
            ]
        ],
    ],
];
```

**Comment by beowulfenator** [comment]
There are cases when you don't need to use the same filter in query and aggregation. As for nested aggs, I share your concern. Unfortunately, we're stuck with the old definition of the method, which we can't change without breaking bc. There is a discussion about it in issues. If you have suggestions, feel free to express them. It's nice to get feedback from someone actually using this component.

**Comment by beowulfenator** [comment]
Solved by #92.
